File: app/src/main/java/com/mgaetan89/showsrage/helper/OkHttpGlideModule.java
package com.mgaetan89.showsrage.helper;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.bumptech.glide.Glide;
import com.bumptech.glide.integration.okhttp.OkHttpUrlLoader;
import com.bumptech.glide.load.model.GlideUrl;
import com.mgaetan89.showsrage.network.SickRageApi;
import com.squareup.okhttp.OkHttpClient;

import java.io.InputStream;

public class OkHttpGlideModule extends com.bumptech.glide.integration.okhttp.OkHttpGlideModule {
	@Override
	public void registerComponents(Context context, Glide glide) {
		SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
		boolean selfSignedCertificate = preferences.getBoolean("self_signed_certificate", false);
		OkHttpClient client = SickRageApi.getInstance().getOkHttpClient(selfSignedCertificate);

		glide.register(GlideUrl.class, InputStream.class, new OkHttpUrlLoader.Factory(client));
	}
}

File: app/src/main/kotlin/com/mgaetan89/showsrage/helper/OkHttpGlideModule.kt
package com.mgaetan89.showsrage.helper

import android.content.Context
import com.bumptech.glide.Glide
import com.bumptech.glide.Registry
import com.bumptech.glide.integration.okhttp.OkHttpGlideModule
import com.bumptech.glide.integration.okhttp.OkHttpUrlLoader
import com.bumptech.glide.load.model.GlideUrl
import com.mgaetan89.showsrage.extension.getPreferences
import com.mgaetan89.showsrage.extension.useSelfSignedCertificate
import com.mgaetan89.showsrage.network.SickRageApi
import java.io.InputStream

class OkHttpGlideModule : OkHttpGlideModule() {
    override fun registerComponents(context: Context, glide: Glide, registry: Registry) {
        super.registerComponents(context, glide, registry)

        val selfSignedCertificate = context.getPreferences().useSelfSignedCertificate()
        val client = SickRageApi.instance.getOkHttpClient(selfSignedCertificate)

        registry.append(GlideUrl::class.java, InputStream::class.java, OkHttpUrlLoader.Factory(client))
    }
}
